dataland-frontend/tests/e2e/fixtures/eutaxonomy/financials/EuTaxonomyDataForFinancialsFixtures.ts

```diff
@@ -7,126 +7,113 @@ import {
   type InsuranceKpis,
   type InvestmentFirmKpis,
 } from "@clients/backend";
-import { generateDatapointOrNotReportedAtRandom } from "@e2e/fixtures/common/DataPointFixtures";
 import { generateEuTaxonomyWithBaseFields } from "@e2e/fixtures/eutaxonomy/EuTaxonomySharedValuesFixtures";
-import { type ReferencedDocuments } from "@e2e/fixtures/FixtureUtils";
-import { randomPercentageValue } from "@e2e/fixtures/common/NumberFixtures";
-import { assertDefined } from "@/utils/TypeScriptUtils";
+import { DEFAULT_PROBABILITY, Generator } from "@e2e/utils/FakeFixtureUtils";
+import { generatePercentageValue } from "@e2e/fixtures/common/NumberFixtures";
 
 /**
- * Generates random insurance company KPIs
- * @param referencedReports he reports that can be referenced as data sources
- * @returns random insurance company KPIs
+ * Generates a single eutaxonomy-financials fixture
+ * @param undefinedProbability the probability (as number between 0 and 1) for "undefined" values in nullable fields
+ * @returns a random eutaxonomy-financials fixture
  */
-export function generateInsuranceKpis(referencedReports: ReferencedDocuments): InsuranceKpis {
-  const taxonomyEligibleNonLifeInsuranceActivities = randomPercentageValue();
-  return {
-    taxonomyEligibleNonLifeInsuranceActivities: generateDatapointOrNotReportedAtRandom(
-      taxonomyEligibleNonLifeInsuranceActivities,
-      referencedReports,
-    ),
-  };
+export function generateEuTaxonomyDataForFinancials(
+  undefinedProbability = DEFAULT_PROBABILITY,
+): EuTaxonomyDataForFinancials {
+  const dataGenerator = new EuFinancialsGenerator(undefinedProbability);
+  return dataGenerator.generateEuTaxonomyDataForFinancialsWithTypes();
 }
 
-/**
- * Generates random credit institution KPIs
- * @param referencedReports he reports that can be referenced as data sources
- * @returns random credit institution KPIs
- */
-export function generateCreditInstitutionKpis(referencedReports: ReferencedDocuments): CreditInstitutionKpis {
-  let tradingPortfolioAndInterbankLoans = undefined;
-  let interbankLoans = undefined;
-  let tradingPortfolio = undefined;
+export class EuFinancialsGenerator extends Generator {
+  financialServicesTypes: EuTaxonomyDataForFinancialsFinancialServicesTypesEnum[];
 
-  const singleOrDualField = faker.datatype.boolean();
-  if (singleOrDualField) {
-    tradingPortfolioAndInterbankLoans = randomPercentageValue();
-  } else {
-    interbankLoans = randomPercentageValue();
-    tradingPortfolio = randomPercentageValue();
+  constructor(undefinedProbability = DEFAULT_PROBABILITY) {
+    super(undefinedProbability);
+    this.financialServicesTypes = faker.helpers.arrayElements(
+      Object.values(EuTaxonomyDataForFinancialsFinancialServicesTypesEnum),
+    );
```

Comment:
When I saw this, it rang a bell in my head:
You want to pick a random number of elements from an array of elements here and you don't want to pick the same element twice. 
This use case is actually required at many places in our fixtures code, but it is not done consistently.

That's why I picked all the places where we could use this `.arrayElements()` functionality from the `faker` library to simplify this use case.
I also looked at places where we might use `.arrayElement()` (without "s") which picks one specific element randomly:

- `getRandomCompanyLegalForm()` => Here we could simply do `arrayElement()` on the available legalForms. Also this should be named `generate...` per our new naming convention right?
- `generateReportingPeriod()` => same thing as for `getRandomCompanyLegalForm()`
- `generateIso4217CurrencyCode()` => again same thing

- `generateListOfRandomNaceCodes()` => Code is too complex for the use case.  Simply do `.arrayElements()` on the availabe Nace Codes?  And why only "A" and "B" ?  The max-number says "5", so the availabe Nace Codes should at least have five different options.

- `getRandomNumberOfDistinctElementsFromArray()` => Check out the usage of this util-function. I think it can easily be replaced by a usage of `arrayElements()`. Then we could delete this custom util-function.

Reply:
done

Reply:
Ausstehende Punkte:
In SmePreparedFixtures.ts braucht das Fixture "Sme-Year..." in seinem "sector" Feld immer mindestens 2 fixtures.
+
Refactoring von "generateListOfRandomNaceCodes".

Reply:
overlooked those, also changed now